src/pages/AddVacancy/index.tsx

```diff
@@ -1,15 +1,123 @@
-import React from "react";
+import React, { useState } from "react";
 import { LuBold } from "react-icons/lu";
 import { FaItalic } from "react-icons/fa";
 import { RiUnderline } from "react-icons/ri";
 import { MdOutlineFormatListNumbered } from "react-icons/md";
 import { MdOutlineFormatListBulleted } from "react-icons/md";
+import { ToastContainer, toast } from "react-toastify";
+import "react-toastify/dist/ReactToastify.css";
+import axios from "axios";
+import { API } from "../../API";
+import { useNavigate } from "react-router-dom";
 
 const AddVacancy = () => {
+  const nav = useNavigate();
+  const [vacancyData, setVacancyData] = useState({
+    type: "", // done
+    currency: "", // done
+    price_from: "", // done
+    price_to: "", // done
+    position: "", // done
+    city: "", // done
+    salary: "", // done
+    description: "", // done
+    email: "", // done
+    organization_name: "", // done
+    organization_icon: "", // done
+  });
+
+  const inputChangeHandler = (e: React.ChangeEvent<HTMLInputElement>) => {
+    let currentInput = e.target.name;
+    setVacancyData({
+      ...vacancyData,
+      [currentInput]: e.target.value,
+    });
+  };
+
+  const textareaChangeHandler = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
+    let currentInput = e.target.name;
+    setVacancyData({
+      ...vacancyData,
+      [currentInput]: e.target.value,
+    });
+  };
+
+  const selectChangeHandler = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    let currentInput = e.target.name;
+    setVacancyData({
+      ...vacancyData,
+      [currentInput]: e.target.value,
+    });
+  };
+
+  const handleAddVacancy = async (e: React.FormEvent) => {
+    e.preventDefault();
+    if (
+      vacancyData.city.trim() === "" ||
+      vacancyData.email.trim() === "" ||
+      vacancyData.organization_name.trim() === "" ||
+      vacancyData.organization_icon.trim() === "" ||
+      vacancyData.price_from.trim() === "" ||
+      vacancyData.position.trim() === "" ||
+      vacancyData.price_to.trim() === ""
+    ) {
+      toast.error("Please fill the empty fields!", {
+        position: "top-center",
+        autoClose: 3000,
+        hideProgressBar: false,
+        closeOnClick: true,
+        pauseOnHover: true,
+        draggable: true,
+        progress: undefined,
+        theme: "colored",
+      });
```

Comment:
util'ge jildirip, tomonkudoi kil
`toast('Please fill the empty fields", "error");`